apache/datadog_checks/apache/data/conf.yaml.example

```diff
@@ -1,46 +1,99 @@
+
+    ## @param apache_user - optional - string
+    ## <USERNAME> for the Apache status endpoint authentication.
+    #
+    # apache_user: <USERNAME>
+
+    ## @param apache_password - optional - string
+    ## <PASSWORD> for the Apache status endpoint authentication.
+    #
+    # apache_password: <PASSWORD>
+
+    ## @param tags - optional - list of key:value element
+    ## List of tags to attach to every metrics, events and service checks emitted by this integration.
+    ## Learn more about tagging: https://docs.datadoghq.com/tagging/
+    #
     # tags:
-    #   - optional_tag
+    #   - <KEY>:<VALUE>
 
-    # The (optional) disable_ssl_validation will instruct the check
-    # to skip the validation of the SSL certificate of the URL being tested.
-    # Defaults to false, set to true if you want to disable SSL certificate validation.
+    ## @param disable_ssl_validation - optional - boolean - default: false
+    ## Instructs the check to skip the validation of the SSL certificate of the URL being tested.
+    ## Defaults to false, set to true if you want to disable SSL certificate validation.
     #
     # disable_ssl_validation: false
 
-    # The (optional) connect_timeout will override the default value, and fail
-    # the check if the time to establish the (TCP) connection exceeds the
-    # connect_timeout value (in seconds)
-    # connect_timeout: 5
+    ## @param connect_timeout - optional - integer 
+    ## Overrides the default connection timeout value, and fails the check if the time to establish the (TCP) connection 
+    ## exceeds the connect_timeout value (in seconds)
+    #
+    # connect_timeout: <VALUE_IN_SECOND>
     
-    # The (optional) receive_timeout will override the default value, and fail
-    # the check if the time to receive the server status from the Apache server
-    # exceeds the receive_timeout value (in seconds)
-    # receive_timeout: 15
+    ## @param receive_timeout - optional - integer 
+    ## Overrides the default received timeout value, and fails the check if the time to receive 
+    ## the server status from the Apache server exceeds the receive_timeout value (in seconds)
+    #
+    # receive_timeout: <VALUE_IN_SECOND>
     
 
 ## Log Section (Available for Agent >=6.0)
-
+#
 #logs:
 
-    # - type : (mandatory) type of log input source (tcp / udp / file)
-    #   port / path : (mandatory) Set port if type is tcp or udp. Set path if type is file
-    #   service : (mandatory) name of the service owning the log
-    #   source : (mandatory) attribute that defines which integration is sending the logs
-    #   sourcecategory : (optional) Multiple value attribute. Can be used to refine the source attribtue
-    #   tags: (optional) add tags to each logs collected
+    ## @param type - mandatory - string
+    ## Type of log input source, to choose between tcp / udp / file / journald / docker
+    #    
+    # - type: file
+
+    ## @param path - mandatory - string
```

Comment:
Why do we need the @param here?

Reply:
The idea to ease parsing of such files. This will open doors down the road (whether we want to do something with it or not) For example to:
- Validate the config.
- Generate documentation in our documentation.
- We could even imagine having a tool user could use to create their configs.
- If not by us, this could be used by the community itself to do such things :)

Reply:
I agree with @irabinovitch here in that it creates unnecessary visual bloat. Parsing is still easy; simply find the first `##` after a new line.

Reply:
The `@param` was meant to add robustness to the parser, there's a discussion in the internal RFC with more details. If we assume a param cannot exceed one line, the description would be "all the lines starting with `##` following the param definition". In general, I would not use blanks as delimiters for a tokenizer processing a configuration file that would look like plain text to it (since we're using comments, not yaml to deliver contents, the less assumptions about the format, the better).

Another use case off the top of my head, if you only need to collect the parameters, by using annotations you'd only need a very simple regex.

Reply:
IMHO, regardless of parsing (which can be done with or without `@param` in the end), I just find the file easier to read. Having `param` create visual "sections".
Regarding the description, every line after the `@param` line up to the first line without comment or with just one `#` is a part of the param description. At least, that's how I understand it.